app/queries/servers/preference.ts

```diff
@@ -65,3 +67,17 @@ export const deletePreferences = async (database: ServerDatabase, preferences: P
         return false;
     }
 };
+
+export const queryPreferencesByCategory = (database: Database, category: string, name?: string, value?: string) => {
+    const clauses = [Q.where('category', category)];
+
+    if (typeof name === 'string') {
```

Comment:
I like better the `!= null` option to make sure it is not undefined, but probably just personal preference.

Reply:
`typeof` covers all three cases; skipping `name = null` and `name = undefined` but still passing true for `name = ''`.

Reply:
`!= null` also cover the three cases: https://stackoverflow.com/questions/38977829/what-is-the-difference-between-null-and-null
(see that it is just one `=` and not two.